From 6a5dacf4f145103fda49cebd2a1ae3dbb60155da Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 7 Jul 2021 12:09:38 +0200
Subject: [PATCH] Add TBB::tbb to TBB_IMPORTED_TARGETS

---
 cmake/TBB.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/TBB.cmake b/cmake/TBB.cmake
index 3723d0d..448ad05 100644
--- a/cmake/TBB.cmake
+++ b/cmake/TBB.cmake
@@ -27,6 +27,7 @@ macro(handle_tbb_target)
     if(TBB_FOUND)
         set_property(TARGET TBB::tbb PROPERTY "MAP_IMPORTED_CONFIG_RELWITHMDD" "DEBUG")
         include_directories_with_host_compiler(${_tbb_include_dirs})
         list(APPEND TBB_IMPORTED_TARGETS TBB::tbb)
+        list(APPEND EXTRA_SHARED_LIBS ${TBB_IMPORTED_TARGETS})
 
         # Print TBB location
-- 
2.30.2

