commit 0e3e10853d432c55fc0a949736576c99d68aee0d
Author: Isuru Fernando <isuruf@gmail.com>
Date:   Sun Nov 20 14:22:59 2022 -0600

    Do not let altivec.h redefine bool

diff --git a/include/oneapi/dnnl/dnnl_ocl.h b/include/oneapi/dnnl/dnnl_ocl.h
index c2478f131..a6e26d496 100644
--- a/include/oneapi/dnnl/dnnl_ocl.h
+++ b/include/oneapi/dnnl/dnnl_ocl.h
@@ -28,6 +28,9 @@
 #endif
 
 #include <CL/cl.h>
+#if defined( __VEC__ ) && defined(bool)
+#undef bool
+#endif
 /// @endcond
 
 #ifdef __cplusplus
diff --git a/src/cpu/ppc64/ppc64_gemm_s8x8s32.hpp b/src/cpu/ppc64/ppc64_gemm_s8x8s32.hpp
index d61331cae..14da1ab1f 100644
--- a/src/cpu/ppc64/ppc64_gemm_s8x8s32.hpp
+++ b/src/cpu/ppc64/ppc64_gemm_s8x8s32.hpp
@@ -15,6 +15,9 @@
 *******************************************************************************/
 
 #include <altivec.h>
+#ifdef bool
+#undef bool
+#endif
 #include "cpu/simple_q10n.hpp"
 
 namespace dnnl {
