diff --git a/cmake/TBB.cmake b/cmake/TBB.cmake
index 2a7d8dc55..5dd430686 100644
--- a/cmake/TBB.cmake
+++ b/cmake/TBB.cmake
@@ -34,6 +34,7 @@ endif()
 find_package_tbb(REQUIRED)
 if(TBB_FOUND)
     include_directories(${_tbb_include_dirs})
+    list(APPEND TBB_IMPORTED_TARGETS TBB::tbb)
     list(APPEND EXTRA_SHARED_LIBS ${TBB_IMPORTED_TARGETS})
 
     # Print TBB location
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 014b8693a..07d2906c8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -60,6 +60,7 @@ if(DNNL_ENABLE_JIT_PROFILING OR DNNL_ENABLE_ITT_TASKS)
     if (UNIX AND NOT APPLE)
         # Not every compiler adds -ldl automatically
         list(APPEND EXTRA_SHARED_LIBS "${CMAKE_DL_LIBS}")
+        list(APPEND EXTRA_SHARED_LIBS "-lrt")
     endif()
 endif()
 
